openff/interchange/_tests/unit_tests/operations/test_combine.py

```diff
@@ -190,3 +190,29 @@ def test_mix_different_5_6_rounding(self, parsley, sage, ethanol):
             ).combine(
                 sage.create_interchange(ethanol.to_topology()),
             )
+
+    @pytest.mark.parametrize("flip_order", [False, True])
+    def test_constraint_key_collision(self, parsley, sage, ethanol, flip_order):
+        """Test that key collisions in constraints are handled."""
+        interchanges = [
+            parsley.create_interchange(
+                ethanol.to_topology(),
+            ),
+            sage.create_interchange(
+                ethanol.to_topology(),
+            ),
+        ]
+
+        # want to make sure this behavior is not order-dependent
+        if flip_order:
+            interchanges.reverse()
+
+        constraint_arrays = [interchange["Constraints"].get_system_parameters() for interchange in interchanges]
```

Comment:
(not blocking) as a sanity check, you might assert that the two interchanges' constraint parameters AREN'T identical. I'm nearly certain that Sage and Parsley have slightly different values for these bond lengths but this could make the test more resilient.